scripts/release_notes.rb

```diff
@@ -7,57 +7,89 @@
 require 'uri'
 require 'net/http'
 require 'json'
-require 'pry-byebug'
 
 URL = 'https://projects.theforeman.org'
 
-# Check out installer modules
-# Compare packages in nightly vs release and retag dependencies
-
-# Create new release via API call
-# Create new search by open + release field
-
-unless @current_release_id = ARGV[0]
-  puts "Usage: #{$0} RELEASE_ID"
-  puts "Hint: 1.16.0 is 240"
+unless ARGV.size == 2
+  puts "Usage: #{$0} PROJECTS RELEASE_NAME"
+  puts "Example: #{$0} foreman,puppet-foreman,selinux,smart-proxy 1.18.0"
   exit 1
 end
 
-# Move bugs with Release set to VERSION to NEXT-VERSION with status new
-def gather_issues(offset=0)
-  url = "#{URL}/projects/foreman/issues.json?status_id=closed&offset=#{offset}&limit=100&release_id=#{@current_release_id}"
+@project_names = ARGV[0].split(',')
+@current_release_name = ARGV[1]
+
+def url_to_json(url)
   uri = URI(URI.escape(url))
   response = Net::HTTP.get(uri)
-	result = JSON.parse(response)
-	if result['total_count'].to_i - offset - 100 <= 0
-		result['issues']
-	else
-		result['issues'] += gather_issues(offset + 100)
-	end
+  JSON.parse(response)
 end
 
-def modify_target_version!(issue_id, options)
-  uri = URI(URI.escape("#{URL}/issues/#{issue_id}.json"))
-  req = Net::HTTP::Put.new(
-		uri,
-		{ 'Content-Type' => 'application/json',
-		  'X-Redmine-API-Key' => File.read('~/.redminekey') }
-	)
-  Net::HTTP.new(uri.host, uri.port).start {|http| http.request(req) }
+# Determine version
+def get_version_id(project)
+  url = "#{URL}/projects/#{project}/versions.json"
+  result = url_to_json(url)
+  return nil if result['versions'].nil?
+  result = result['versions'].detect{|v| v['name'] == @current_release_name }
+  return nil if result.nil?
+  result['id']
+rescue
+  puts "Error getting version list from #{url}"
+  exit 1
 end
 
-grouped_issues = gather_issues.group_by { |issue| issue['category']['name'] rescue 'Uncategorized' }
+@current_release_ids = @project_names.map{|project| get_version_id(project)}.compact
 
-grouped_issues = Hash[ grouped_issues.sort_by { |key, val| key } ]
-grouped_issues.each do |category, issues|
-	puts "#### #{category}"
+if @current_release_ids.empty?
+  puts "Release #{@current_release_name} not found in any of the projects #{@project_names}"
+  exit 1
+end
 
-  issues.each do |issue|
-	  puts "* #{issue['subject']} ([##{issue['id']}](#{URL}/issues/#{issue['id']}))"
+def releases_filter
+  @current_release_ids.map{|release| "v[cf_12][]=#{release}"}.join('&')
+end
+
+def gather_issues(offset = 0, feature = true)
+  url = "#{URL}/projects/foreman/issues.json?status_id=closed&offset=#{offset}&limit=100&f[]=cf_12&op[cf_12]==&#{releases_filter}&f[]=tracker_id&op[tracker_id]=#{feature ? '=':'!'}&v[tracker_id][]=2"
```

Comment:
Maybe we should store the parameters in a hash and use `URI.encode_www_form`. Probably easier to understand. You also have a `op[cf_12]==` but is that intended?

Reply:
`URI.encode_www_form` doesn't seem to handle arrays in the way rails expects them, but maybe i'm doing it wrong:
```
[5] pry(main)> URI.encode_www_form({f: ['cf_12', 'status'], op: {cf_12: '='}, v: {cf_12: [1,2]}})
=> "f=cf_12&f=status&op=%7B%3Acf_12%3D%3E%22%3D%22%7D&v=%7B%3Acf_12%3D%3E%5B1%2C+2%5D%7D"
```
`op[cf_12]==` is intended, but i think the second = should be escaped - it indicates which action you're matching on (= - is, ! - isn't etc)

Reply:
actually i take it back, if i escape it first it doesn't work, guessing double escape is breaking it